### DISCUSSION TITLE
How to print dynamically created columns in q-table ?

### DISCUSSION THREAD

**Discussion by ewertonrfd** [quasarframework/quasar#14089] [General - Components / Directives / etc]
Hi everyone,

4 of the columns of the table I already know and I put the keys to be able to edit these fields
But the other columns will be dynamic, the person will put any name, but I won't need to edit the values, they will be fixed, but I need to show them in the table

<img width="505" alt="3 td eu conheco os outros nao" src="https://user-images.githubusercontent.com/13201575/181674559-8249bcf5-3b89-4c4c-94d8-8b43364e1625.png">

The detail is just to show the column text

How to make? I tried in some ways but still no result, thanks in advance

**Comment by MilosPaunovic** [comment]
What I can gather from your explanation is that you need to define that extra column in you `columns` array and use its name as a key for `q-td` in `body` slot. If that does not help you, please provide a Codepen reproduction of your issue.

**Reply by ewertonrfd** [reply]
Hello @MilosPaunovic ,
If I set it in columns it will work!
But I need this to be dynamic, the user will define the other columns (not me)
The idea is how to show in this table that it is already working, only with these dynamic columns

I'm testing something with the body-cell-[name] slot

**Reply by EstefanyEH** [reply]
Hi @EwertonDutra , I'm trying to do the same but it didn't work, did you?

**Reply by ewertonrfd** [reply]
Hello,
Yes, I did it, but it wasn't how I wanted it.
but it worked 😁

**Reply by ewertonrfd** [reply]
It was like this:
<img width="503" alt="Captura de tela 2022-08-03 121559" src="https://user-images.githubusercontent.com/13201575/182644837-cbde5eef-a8b3-4218-ab3d-d20d0bde62db.png">
The documentation already shows this.
What I did differently was:
<img width="256" alt="Captura de tela 2022-08-03 121731" src="https://user-images.githubusercontent.com/13201575/182645155-d03d9ef0-f6d0-4cab-9093-90bfb15047cd.png">
The columns that are dynamic I feed through a dialog (modal)
<img width="232" alt="Captura de tela 2022-08-03 121853" src="https://user-images.githubusercontent.com/13201575/182645493-ea284569-bd16-41be-9213-f706dcd36b71.png">

Since I can't make the columns dynamic (at least as far as I know and I was able to search), I did it myself 😅
**Hope it helps too, who knows!**
